warehouse/client/warehouse.go

```diff
@@ -0,0 +1,141 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/utils/httputil"
+)
+
+const defaultTimeout = 10 * time.Second
+
+// StagingFile contains the require metadata to process a staging file.
+type StagingFile struct {
```

Comment:
Since we are using this. Can we remove the StagingFile struct in [here](https://github.com/rudderlabs/rudder-server/blob/b7aa80003e237ddda370dff125898d4ac52ea820/warehouse/utils/utils.go#L266).